build/_build/Honeypot/IntegrationGroups.cs

```diff
@@ -0,0 +1,149 @@
+            NugetPackages.Add("RabbitMQ.Client", new string[] { "RabbitMQ.Client" });
+            NugetPackages.Add("ServiceStack.Redis", new string[] { "ServiceStack.Redis" });
+            NugetPackages.Add("System.Data.SqlClient", new string[] { "System.Data.SqlClient" });
+            NugetPackages.Add("Microsoft.Data.SqlClient", new string[] { "Microsoft.Data.SqlClient" });
+            NugetPackages.Add("Microsoft.Data.Sqlite", new string[] { "Microsoft.Data.Sqlite" });
+            NugetPackages.Add("System.Data.SQLite", new string[] { "System.Data.SQLite" });
+            NugetPackages.Add("StackExchange.Redis", new string[] { "StackExchange.Redis" });
+            NugetPackages.Add("StackExchange.Redis.StrongName", new string[] { "StackExchange.Redis.StrongName" });
+            NugetPackages.Add("System.ServiceModel", new string[] { "System.ServiceModel.Http" });
+            NugetPackages.Add("System.Net.Requests", new string[] { "System.Net.Requests" });
+            NugetPackages.Add("xunit.execution.dotnet", new string[] { "xunit.extensibility.execution" });
+            NugetPackages.Add("xunit.execution.desktop", new string[] { });
+        }
+
+        private IntegrationMap()
+        { 
+        }
+
+        public static async Task<IntegrationMap> Create(string name, string assemblyName, Version maximumVersion)
+        {
+            var instance = new IntegrationMap();
+
+            if (!NugetPackages.ContainsKey(name))
+            {
+                throw new Exception($"Missing key: {name} - Every integration must be represented in the packages map.");
+            }
+
+            instance.Name = name;
+            instance.AssemblyName = assemblyName;
+            instance.MaximumAssemblyVersion = maximumVersion;
+
+            await instance.PopulatePackages();
+
+            return instance;
+        }
+
+        public string Name { get; set; }
+
+        public string AssemblyName { get; set; }
+
+        public Version MaximumAssemblyVersion { get; set; }
+
+        public List<IntegrationPackage> Packages { get; set; } = new List<IntegrationPackage>();
+
+        private async Task PopulatePackages()
+        {
+            var packageNames = NugetPackages[Name];
+            foreach (var packageName in packageNames)
+            {
+                var searchCriteria = new PackageSearchCriteria
+                {
+                    IntegrationName = Name,
+                    NugetPackageSearchName = packageName,
+                    MinVersion = "0.0.1",
+                    MaxVersionExclusive = "255.255.255"
+                };
+
+                var packages = await NuGetPackageHelper.GetPackageMetadatas(searchCriteria);
+
+                var potentiallySupportedPackages = packages.Where(p => p.Identity.HasVersion).OrderByDescending(p => p.Identity.Version);
+
+                // TODO: Download and check referenced assemblies for assembly versions
```

Comment:
I would like to actually check assembly versions in order to know if we support the package. Thoughts?

Reply:
Would be nice, but not a priority IMO. Can't imagine there's many (any?) cases in which our "supported version wildcard" _includes_ the NuGet but _not_ the assembly version (or vice versa). But we won't know until we try I guess 😄